evm/deploy-create3-deployer.js

```diff
@@ -0,0 +1,111 @@
+const implementationJson = require('../artifacts/contracts/deploy/Create3Deployer.sol/Create3Deployer.json');
+const contractName = 'Create3Deployer';
+
+async function deploy(options, chain) {
+    const { env, privateKey, verify } = options;
+    const wallet = new Wallet(privateKey);
+
+    printInfo('Deployer address', wallet.address);
+
+    const rpc = chain.rpc;
+    const provider = getDefaultProvider(rpc);
+
+    console.log(
+        `Deployer has ${(await provider.getBalance(wallet.address)) / 1e18} ${chalk.green(
+            chain.tokenSymbol,
+        )} and nonce ${await provider.getTransactionCount(wallet.address)} on ${chain.name}.`,
+    );
+
+    const contracts = chain.contracts;
+
+    if (!contracts[contractName]) {
+        contracts[contractName] = {};
+    } else {
+        throw new Error(`Create3 deployer is already deployed on ${chain.name} ${env} at address: ${contracts[contractName].address}`);
+    }
+
+    const contractConfig = contracts[contractName];
+    const gasOptions = contractConfig.gasOptions || chain.gasOptions || {};
+    console.log(`Gas override for chain ${chain.name}: ${JSON.stringify(gasOptions)}`);
+
+    const salt = options.salt || contractName;
+    printInfo('Create3 deployer deployment salt', salt);
+
+    const constAddressDeployer = contracts.ConstAddressDeployer.address;
+    const create3DeployerAddress = await predictContractConstant(constAddressDeployer, wallet.connect(provider), implementationJson, salt);
+    printInfo('Create3 deployer will be deployed to', create3DeployerAddress);
+
+    console.log('Does this match any existing deployments?');
+    const anwser = readlineSync.question(`Proceed with deployment on ${chain.name}? ${chalk.green('(y/n)')} `);
+    if (anwser !== 'y') return;
+
+    const contract = await deployCreate2(
+        constAddressDeployer,
+        wallet.connect(provider),
+        implementationJson,
+        salt,
+        [],
+        gasOptions,
+        env,
+        chain.name,
+        verify,
+    );
+
+    contractConfig.salt = salt;
+    contractConfig.address = contract.address;
+    contractConfig.deployer = wallet.address;
+
+    printInfo(`${chain.name} | ConstAddressDeployer:`, constAddressDeployer);
+    printInfo(`${chain.name} | Create3Deployer`, contractConfig.address);
+}
+
+async function main(options) {
+    const config = require(`${__dirname}/../info/${options.env}.json`);
+
+    const chains = options.chainNames.split(',');
+
+    for (const chain of chains) {
+        if (config.chains[chain.toLowerCase()] === undefined) {
+            throw new Error(`Chain ${chain} is not defined in the info file`);
+        }
+    }
+
+    for (const chain of chains) {
+        await deploy(options, config.chains[chain.toLowerCase()]);
+        writeJSON(config, `${__dirname}/../info/${options.env}.json`);
+    }
+}
+
+const program = new Command();
+
+program.name('deploy-create3-deployer').description('Deploy create3 deployer');
```

Comment:
maybe move the main deploy scripts under a `deploy` subfolder so we can distinguish it better, and we don't need a deploy prefix in the file names

Reply:
moved deployment scripts under a deploy subfolder within the evm folder